tox_docker.py

```diff
@@ -195,6 +206,9 @@ def tox_runtest_pre(venv):
         name, _, tag = image.partition(":")
         gateway_ip = _get_gateway_ip(container)
         for containerport, hostports in container.attrs["NetworkSettings"]["Ports"].items():
+            if hostports is None:
```

Comment:
When is this `None`?

Reply:
I should add a comment here. When a container like mysql:5.7 `EXPOSES` a port, but you explicitly do not publish it, the entry still appears, but with `null`.

Reply:
Ah, that makes sense. Yeah, a comment would be a good addition.